worlds/lingo/regions.py

```diff
@@ -0,0 +1,78 @@
+from BaseClasses import MultiWorld, Region, Entrance
+from .static_logic import StaticLingoLogic, Room, RoomEntrance
+from .locations import LingoLocation
+from .player_logic import LingoPlayerLogic
+from .Options import get_option_value
+
+
+def create_region(room: Room, multiworld: MultiWorld, player: int, player_logic: LingoPlayerLogic):
+    new_region = Region(room.name, player, multiworld)
+
+    if room.name in player_logic.LOCATIONS_BY_ROOM.keys():
+        for location in player_logic.LOCATIONS_BY_ROOM[room.name]:
```

Comment:
```suggestion
        new_region.add_locations(player_logic.LOCATIONS_BY_ROOM[room.name], LingoLocation)
```
Could do either this suggestion, or add the access_rules on creation. I think either is fine, but currently you're essentially looping through all the locations twice.